### DISCUSSION TITLE
use_cross_filter with ToggleButtonsMultiple?

### DISCUSSION THREAD

**Discussion by Kugeleis** [widgetti/solara#664] [Q&A] [answered]
I'd prefer ToggleButtonsMultiple over CrossFilterSelect for filtering e.g. a dataframe. Can this be achieved?

**Comment by maartenbreddels** [comment]
Yes, you can write your own. You can take a look at https://github.com/widgetti/solara/blob/d050011c8ca850b9bef6039122672e7fbf886a7e/solara/components/cross_filter.py#L46 and replace the select with the toggle buttons.
Does that make sense?

**Reply by Kugeleis** [reply]
I tried to add `solara.ToggleButtonsMultiple(value=value,values=items,on_value=set_values_and_filter)` after line 138 but I get TypeError: 'NoneType' object is not iterable. I guess _ToggleButtonsMultiple_ expects different data structures than _Select_ .

**Reply by maartenbreddels** [reply, marked as answer]
Indeed, value cannot be None (see https://solara.dev/documentation/components/input/togglebuttons or  https://github.com/widgetti/solara/blob/76ce17b2c9c95f534496e69fb028b3027dde584d/solara/components/togglebuttons.py#L145 )

If you start from an empty list, that would be one thing. Also, if you use mypy or pylance (in vscode) you probably get type hints if you do something wrong (like passing None).

I hope that helps you.
